proc.c

```diff
@@ -37,7 +37,11 @@ const rb_cref_t *rb_vm_cref_in_context(VALUE self, VALUE cbase);
 struct METHOD {
     const VALUE recv;
     const VALUE klass;
+    /* needed for #super_method */
     const VALUE iclass;
+    /* Different than me->owner only for ZSUPER methods.
+       This is error-prone but unavoidable until ZSUPER methods are removed. */
```

Comment:
`s/until/unless/`.  I appreciate your optimism, but from matz's comments, it doesn't sound like we plan to remove ZSUPER methods.